specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json

```diff
@@ -155,6 +267,212 @@
+        }
+      }
+    },
+    "ApiManagementGatewayUpdatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.GatewayUpdated event.",
+      "type": "object",
+      "properties": {
+        "resourceUri": {
+          "description": "The fully qualified ID of the resource that the compliance state change is for, including the resource name and resource type. Uses the format, `/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroup>/Microsoft.ApiManagement/service/<ServiceName>/gateways/<ResourceName>`",
+          "type": "string"
+        }
+      },
+      "x-ms-examples": {
+        "gatewayUpdatedCloudEventsSchema": {
+          "$ref": "./examples/cloud-events-schema/gatewayUpdated.json"
+        },
+        "gatewayUpdatedEventGridSchema": {
+          "$ref": "./examples/event-grid-schema/gatewayUpdated.json"
+        }
+      }
+    },
+    "ApiManagementGatewayDeletedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.GatewayDeleted event.",
+      "type": "object",
+      "properties": {
+        "resourceUri": {
+          "description": "The fully qualified ID of the resource that the compliance state change is for, including the resource name and resource type. Uses the format, `/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroup>/Microsoft.ApiManagement/service/<ServiceName>/gateways/<ResourceName>`",
+          "type": "string"
+        }
+      },
+      "x-ms-examples": {
+        "gatewayDeletedCloudEventsSchema": {
+          "$ref": "./examples/cloud-events-schema/gatewayDeleted.json"
+        },
+        "gatewayDeletedEventGridSchema": {
+          "$ref": "./examples/event-grid-schema/gatewayDeleted.json"
+        }
+      }
+    },
+    "ApiManagementGatewayHostnameConfigurationCreatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.GatewayHostnameConfigurationCreated event.",
+      "type": "object",
+      "properties": {
+        "resourceUri": {
+          "description": "The fully qualified ID of the resource that the compliance state change is for, including the resource name and resource type. Uses the format, `/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroup>/Microsoft.ApiManagement/service/<ServiceName>/gateways/<GatewayName>/hostnameConfigurations/<ResourceName>`",
+          "type": "string"
+        }
+      },
+      "x-ms-examples": {
+        "gatewayHostnameConfigurationCreatedCloudEventsSchema": {
+          "$ref": "./examples/cloud-events-schema/gatewayHostnameConfigurationCreated.json"
+        },
+        "gatewayHostnameConfigurationCreatedEventGridSchema": {
+          "$ref": "./examples/event-grid-schema/gatewayHostnameConfigurationCreated.json"
+        }
+      }
+    },
+    "ApiManagementGatewayHostnameConfigurationUpdatedEventData": {
```

Comment:
I think we will probably get some architect feedback about the length of these names. Is there anything we can do to shorten them?

Reply:
We have already discussed them with Event Grid team and we are following the typical event type naming convention.

If we reduce the length more the names reduce customer experience

Reply:
The Event Grid team that approved it (@robece, @jfggdl & @Kishp01) agreed on the shorter naming that we introduced here (instead of #20871).

Since there is no rule on the length of event types we've agreed there is no problem, but @robece will look at introducing length limits for future events.

We are rolling this out as we speak across all regions because of the earlier sign off so @maksimkim & myself are hoping these new events will not be blocked.

Reply:
These events should not be rolled out until the Azure SDK architecture review board has had a chance to review them.

Reply:
/cc @lmazuel

Reply:
The changes are live in APIM but not enabled in Event Grid. The reason they were already rolled out is that they have to be verified in EG in canary before opening this PR as per EG guidelines.

It's OK to review them now since they are not enabled yet so no worries.

I am not able to join that meeting (as per my email) since it is not EU friendly so it would be nice to reschedule it.

However, as mentioned above they are already reviewed by EG team.